lib/trip_dispatcher.rb

```diff
@@ -25,52 +31,146 @@ def load_users(filename)
 
         users << User.new(input_data)
       end
-
       return users
     end
 
-
     def load_trips(filename)
       trips = []
-      trip_data = CSV.open(filename, 'r', headers: true,
-                                          header_converters: :symbol)
+      trip_data = CSV.open(filename, headers: true, header_converters: :symbol)
 
       trip_data.each do |raw_trip|
-        passenger = find_passenger(raw_trip[:passenger_id].to_i)
-
+        driver = find_driver(raw_trip[:driver_id].to_i) #instance of Driver class
+        passenger = find_passenger(raw_trip[:passenger_id].to_i) # instance of User class
         parsed_trip = {
           id: raw_trip[:id].to_i,
-          passenger: passenger,
-          start_time: raw_trip[:start_time],
-          end_time: raw_trip[:end_time],
+          driver: driver.id,
+          passenger: passenger.id,
+          start_time: Time.parse(raw_trip[:start_time]),
+          end_time: Time.parse(raw_trip[:end_time]),
           cost: raw_trip[:cost].to_f,
           rating: raw_trip[:rating].to_i
         }
 
         trip = Trip.new(parsed_trip)
-        passenger.add_trip(trip)
-        trips << trip
+        passenger.add_trip(trip) # User.add_trip(trip) adds to @trips []
+        driver.add_driven_trip(trip) # Driver.add_trip(trip) adds to @trips []
+        trips << trip # TripDispatcher.load_trips(filename) adds to trips [], which is returned by load_trips. init loads trips to TripDisp @trips
       end
 
       return trips
     end
 
+    def load_drivers(filename)
+      drivers = []
+
+      CSV.read(filename, headers: true).each do |line|
+        input_data = {}
+
+        user = find_passenger(line[0].to_i)
+
+        input_data[:id] = line[0].to_i
+        input_data[:vehicle_id] = line[1]
+        input_data[:status] = line[2].to_sym
+
+        input_data[:name] = user.name
+        input_data[:phone_number] = user.phone_number
+        input_data[:trips] = []#user.trips
+
+        drivers << Driver.new(input_data)
+      end
+      return drivers
+    end
+
+    def find_driver(id)
+      check_id(id)
+      found_driver = @drivers.find { |driver| driver.id == id } #instance of Driver class
+      if found_driver == nil
+        raise ArgumentError, "Searched for driver #{id} but none found"
+      else
+        return found_driver
+      end
+    end
+
+    def available_driver(user_id)
+      check_id(user_id)
+      time_at_delay = Time.now
+      begin
+        available = @drivers.find { |driver|
+          driver.status == :AVAILABLE && driver.id != user_id
+        }
+        if available == nil
```

Comment:
`available` can't be `nil` instead you should check to see if `available.empty?`